[ADP-3472] Bump node dependencies to 10.1.1

paolino opened the pull request:
- Bump ouroboros-network dep to 0.17.1.2
- Bump node runtime version to 10.1.1

ADP-3472

paolino committed:
Bump compile time deps ouroboros-network to 0.17.1.2

paolino committed:
Bump node runtime dependency version to 10.1.1

paolino committed:
Add new txSupplementalData field to mkUnsignedTx

paolino committed:
Fix BalanceSpec imported name conflicts among cardano api modules

Anviking commented on lib/cardano-api-extra/lib/Cardano/Api/Gen.hs:
```diff
@@ -1760,6 +1760,19 @@ genVotingProcedures w = case w of
+    $ \case
+        ConwayEraOnwardsConway -> oneof
+            [ pure ShelleyApi.TxSupplementalDataNone
+            , pure $ ShelleyApi.TxSupplementalDatums []
```
Have not tried to compile this, but think it should just be this:
```suggestion
            , ShelleyApi.TxSupplementalDatums <$> genHashableScriptData
```
Otherwise a TODO would be nice given the `[]` and generator name.

paolino replied:
fixed

Anviking reviewed (approved):
lgtm, thanks!

paolino committed:
Remove sanchonet restoration steps

They are not completing ATM, no real value as Conway is already in mainnet

paolino committed:
Remove private restoration steps

Not completing ATM, again no real value in running a restoration there as Conway is on mainnet

paolino committed:
Remove shell command logs from CI docker run script
